### PR TITLE
cleanup monitor.py

### PR DESCRIPTION
**Pull request by A-UNDERSCORE-D**

### CODE REVIEW

**Comment by A-UNDERSCORE-D** [comment]
This is ready ready for testing and any reviews people want to throw at it

File: monitor.py [hunk 1]
```diff
@@ -52,9 +51,10 @@ class EDLogs(FileSystemEventHandler):
     _POLL = 1		# Polling is cheap, so do it often
     _RE_CANONICALISE = re.compile(r'\$(.+)_name;')
     _RE_CATEGORY = re.compile(r'\$MICRORESOURCE_CATEGORY_(.+);')
+    _RE_LOGFILE = re.compile(r'^Journal(Beta)?\.[0-9]{12}\.[0-9]{2}\.log$')
 
     def __init__(self):
-        FileSystemEventHandler.__init__(self)	# futureproofing - not need for current version of watchdog
+        FileSystemEventHandler.__init__(self)  # futureproofing - not need for current version of watchdog
```
Comment (Athanasius) [inline monitor.py:57]: I wonder how old that comment is and if it's no longer futureproofing....

File: monitor.py [hunk 2]
```diff
@@ -71,7 +71,7 @@ def __init__(self):
         # If 3 we need to inject a special 'StartUp' event since consumers won't see the LoadGame event.
         self.live = False
 
-        self.game_was_running = False	# For generation the "ShutDown" event
+        self.game_was_running = False  # For generation the "ShutDown" event
```
Comment (Athanasius) [inline monitor.py:74]: generation **of** the

File: monitor.py [hunk 3]
```diff
@@ -226,80 +250,102 @@ def worker(self):
...
+
             else:
-                self.event_queue.append(None)	# Generate null event to update the display (with possibly out-of-date info)
+                # Generate null event to update the display (with possibly out-of-date info)
+                self.event_queue.append(None)
                 self.live = False
 
         # Watchdog thread
-        emitter = self.observed and self.observer._emitter_for_watch[self.observed]	# Note: Uses undocumented attribute
+        emitter = self.observed and self.observer._emitter_for_watch[self.observed]  # Note: Uses undocumented attribute
```
Comment (Athanasius) [inline monitor.py:272]: I wonder if there's a documented method to achieve this now ?

File: monitor.py [hunk 4]
```diff
@@ -355,54 +404,61 @@ def parse_entry(self, line):
...
-                    'Credits'      : entry['Credits'],
-                    'FID'          : entry.get('FID'),	# From 3.3
-                    'Horizons'     : entry['Horizons'],	# From 3.0
-                    'Loan'         : entry['Loan'],
-                    'Engineers'    : {},
-                    'Rank'         : {},
-                    'Reputation'   : {},
-                    'Statistics'   : {},
-                    'Role'         : None,
+                # Don't set Ship, ShipID etc since this will reflect Fighter or SRV if starting in those
```
Comment (Athanasius) [inline monitor.py:407]: You completely lost the alignment on this one :P

File: monitor.py [hunk 5]
```diff
@@ -411,197 +467,258 @@ def parse_entry(self, line):
...
+
+            elif event_type == 'ModuleBuy':
                 self.state['Modules'][entry['Slot']] = {
-                    'Slot'     : entry['Slot'],
-                    'Item'     : self.canonicalise(entry['BuyItem']),
-                    'On'       : True,
-                    'Priority' : 1,
-                    'Health'   : 1.0,
-                    'Value'    : entry['BuyPrice'],
+                    'Slot': entry['Slot'],
```
Comment (Athanasius) [inline monitor.py:478]: Alignment again

File: monitor.py (hunk 1 above)
Comment (A-UNDERSCORE-D) [reply monitor.py:57]: Yeah I have no idea. on linux FileSystemEventHandler is nonexistent, but given that its "not needed" Im inclined to remove it (or at least replace with `super().__init__()`

File: monitor.py [hunk 6]
```diff
@@ -411,197 +467,258 @@ def parse_entry(self, line):
...
                 payload.pop('timestamp')
-                self.state[entry['event']] = payload
+                self.state[event_type] = payload
+
+            elif event_type == 'EngineerProgress':
+                engineers = self.state['Engineers']
+                if 'Engineers' in entry:  # Startup summary
+                    self.state['Engineers'] = {
+                        e['Engineer']: (e['Rank'], e.get('RankProgress', 0))
+                        if 'Rank' in e else e['Progress'] for e in entry['Engineers']
```
Comment (Athanasius) [inline monitor.py:562]: It feels like the conditional should be indented to emphasise it's attached to the previous line.

File: monitor.py [hunk 7]
```diff
@@ -411,197 +467,258 @@ def parse_entry(self, line):
...
                 module['Engineering'] = {
-                    'Engineer'      : entry['Engineer'],
-                    'EngineerID'    : entry['EngineerID'],
-                    'BlueprintName' : entry['BlueprintName'],
-                    'BlueprintID'   : entry['BlueprintID'],
-                    'Level'         : entry['Level'],
-                    'Quality'       : entry['Quality'],
-                    'Modifiers'     : entry['Modifiers'],
-                    }
+                    'Engineer': entry['Engineer'],
```
Comment (Athanasius) [inline monitor.py:656]: Alignment

File: monitor.py (hunk 3 above)
Comment (A-UNDERSCORE-D) [reply monitor.py:277]: Does not appear to be, went looking

File: monitor.py (hunk 6 above)
Comment (A-UNDERSCORE-D) [reply monitor.py:569]: Oh whoops, I assumed that was part of the comprehension

File: monitor.py (hunk 1 above)
Comment (Athanasius) [reply monitor.py:62]: It's a 'darwin' block of code, and is `watchdog.events.FileSystemEventHandler` which definitely does exist.  But, yeah, using `super().__init__()` is slightly more indicative of the intent.

File: monitor.py (hunk 3 above)
Comment (Athanasius) [reply monitor.py:277]: Add a comment about self.observer.emitters being a Set, and could check for a member with a 'watch' attribute. But as this is working, leave it be for now, we'll address it if watchdog internals ever change.

**Pull request merged**